### ISSUE TITLE
css/css-animations/animation-important-001.html assumes custom properties are not animated discretely

### ISSUE THREAD

**Issue by graouts** [web-platform-tests/wpt#34600]
The test [css/css-animations/animation-important-001.html](https://wpt.fyi/results/css/css-animations/animation-important-001.html?label=experimental&label=master&aligned) assumes that custom properties interpolate non-discretely. For instance:

```javascript
test(() => {
  getComputedStyle(target7).getPropertyValue('--length-1');

  let animation = target7.animate([
    { '--length-1': '100px' },
    { '--length-1': '200px' },
  ], {
    duration: 1000000,
    delay: -500000,
    easing: 'steps(2, end)'
  });

  assert_equals(getComputedStyle(target7).getPropertyValue('--length-1'), '150px');
 , 'Custom property animations appearing via setKeyframes do not override important declarations');
```

But the [CSS Custom Properties for Cascading Variables Module Level 1 specification](https://drafts.csswg.org/css-variables/#defining-variables) clearly states "[Animation type](https://www.w3.org/TR/web-animations/#animation-type): discrete".

This would mean that the assertion above is incorrect and that it should be `200px`.

**Comment by graouts** [comment]
Cc @andruud who authored this test.

**Comment by graouts** [comment]
I do see this other spec text in the [animation behavior](https://drafts.css-houdini.org/css-properties-values-api/#animation-behavior-of-custom-properties) section of [CSS Properties and Values API Level 1](https://drafts.css-houdini.org/css-properties-values-api/):

> When referenced by animations and transitions, custom property values [interpolate](https://drafts.csswg.org/css-values-4/#interpolation) [by computed value](https://drafts.csswg.org/web-animations-1/#by-computed-value), in accordance with the type that they parsed as.

That seems to me to be in contradiction with the previously-cited specification.

However, that seems to be more in line with what an author would expect, it would make sense that a custom property defined as a `<length>` is animatable. It seems to me like it should be possible to let the author define that property as discrete still for cases where a value does not make sense to animate.

**Comment by andruud** [comment]
Yeah, this is intentional. Registered custom properties (can) get a non-discrete animation type.
